src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -213,10 +216,18 @@ class DeepDiveLogParser extends JavaTokenParsers {
     | "[" ~> cond <~ "]"
     )
 
-  def atom = relationName ~ "(" ~ repsep(expr, ",") ~ ")" ^^ {
-    case (r ~ "(" ~ patterns ~ ")") => Atom(r, patterns)
+  def pattern : Parser[Pattern] =
+    ( "_"  ^^ { case _ => PlaceholderPattern() }
+    | expr ^^ {
+        case VarExpr(x) => VarPattern(x)
```

Comment:
Yes, there's ambiguity between parsing into a VarPattern and a VarExpr. So I changed to this one.